java/statestore/src/test/java/sleeper/statestore/dynamodb/DynamoDBStateStoreIT.java

```diff
@@ -812,221 +815,89 @@ public void shouldReturnAllPartitions() throws StateStoreException {
         // Given
         Field field = new Field("key", new LongType());
         Schema schema = Schema.builder().rowKeyFields(field).build();
-        RangeFactory rangeFactory = new RangeFactory(schema);
-        Region region0 = new Region(rangeFactory.createRange(field, Long.MIN_VALUE, 1L));
-        Partition partition0 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region0)
-                .id("id0")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region1 = new Region(rangeFactory.createRange(field, 1L, 100L));
-        Partition partition1 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region1)
-                .id("id1")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region2 = new Region(rangeFactory.createRange(field, 100L, 200L));
-        Partition partition2 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region2)
-                .id("id2")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region3 = new Region(rangeFactory.createRange(field, 200L, null));
-        Partition partition3 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region3)
-                .id("id3")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        StateStore dynamoDBStateStore = getStateStore(schema, Arrays.asList(partition0, partition1, partition2, partition3));
+
+        PartitionTree tree = new PartitionsBuilder(schema).rootFirst("root")
+                .splitToNewChildren("root", "left", "right", 100L)
+                .splitToNewChildren("left", "id1", "id2", 1L)
+                .splitToNewChildren("right", "id3", "id4", 200L).buildTree();
+
+        StateStore dynamoDBStateStore = getStateStore(schema, tree.getAllPartitions());
 
         // When
         List<Partition> retrievedPartitions = dynamoDBStateStore.getAllPartitions();
```

Comment:
This variable isnt needed anymore, and the When/Then sections can be merged together